### ISSUE TITLE
Install kernel dependencies with the -U flag

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#4758]
When install ipykernel, etc we should update the depednencies.
We have had plenty of cases where users had it installed, but versions were incorrect.
They had to uninstall & re-install.

For instance I had an issue with ipykernel in a conda environment.
Had an old version of ipython, hence when installing ipykernel, an old compatible version got installed. However that version of ipython is too old, similarly its possible the versio of ipykernel is also old.

The solution was `conda install ipykernel --update-deps --force-reinstall`
We might want to do the same for `pip install ipykernel -U --force-reinstall`

# How to test/validate
* Install latest dev version of Python extension with the changes (here's the PR in Python ext https://github.com/microsoft/vscode-python/pull/16133)
* Use conda & non conda
* Get notebooks working
* Uninstall pyzmq (a dependency), 
* Now ipykernel is installed, but it doesn't work, hence re-installing ipykernel will not work
* Try now, and it should force re-install

**Comment by DonJayamanne** [comment]
We might also want to consider makign it possible to pass in the argument `--force-reinstall`.
We found at least in 2 occasions that this works.
hence we might have scenarios where we have -U as well as `--force-reinstall`.

TLDR - ensure we can pass more than just the one argument.

**Comment by DonJayamanne** [comment]
This works, see here https://github.com/microsoft/vscode-jupyter/issues/5689  (user tried this option to get unblocked recently)

**Comment by DonJayamanne** [comment]
We should probably use `--update-deps --force-reinstall` only when first install of `ipykernel` fails.
Probably worth discussing with team.

**Comment by DonJayamanne** [comment]
Discussed solution:
* if ipykernel is available (pip list), then pass flags to Python to force update & force re-install

**Comment by joyceerhl** [comment]
I do get a prompt for conda and non conda environments, but for conda the install command fails.

Conda:
![image](https://user-images.githubusercontent.com/30305945/118434606-307e5c00-b692-11eb-9846-0a9d70fc638a.png)

Conda install errors out:
![image](https://user-images.githubusercontent.com/30305945/118434733-6ae7f900-b692-11eb-8da4-4e67847d8e17.png)

Non-conda (pip):
![image](https://user-images.githubusercontent.com/30305945/118434881-abe00d80-b692-11eb-9f3f-cfbce1bbed84.png)

**Comment by DavidKutu** [comment]
validated, in both the jupyter and python extensions
